src/main/java/org/java_websocket/WebSocketImpl.java

```diff
@@ -820,6 +823,19 @@ public WebSocketListener getWebSocketListener() {
 		return (T) attachment;
 	}
 
+	@Override
+	public boolean hasSSLSupport() {
+		return channel instanceof ISSLChannel;
+	}
+
+	@Override
+	public SSLSession getSSLSession() {
+		if (!hasSSLSupport()) {
+			throw new IllegalStateException("This websocket does use ws instead of wss. No SSLSession available.");
```

Comment:
The documentation  and signature of `WebSocket::getSSLSession` states that it throws an `IllegalArgumentException` if SSL is not supported, but the implementation here throws an `IllegalStateException`. I'd say IAE is more appropriate.

Reply:
Also, the message was probably intended to say "This websocket uses ws instead of wss. No SSLSession available."

Reply:
https://github.com/TooTallNate/Java-WebSocket/issues/890#issuecomment-496599139 had the idea that it would be better to throw ISE.

Looking at the documentation of ISE, I would say we should use IAE